apps/admin-portal/src/components/users/add-user.tsx

```diff
@@ -0,0 +1,549 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AlertInterface, AlertLevels } from "@wso2is/core/models";
+import { Field, Forms, Validation } from "@wso2is/forms";
+import { EditSection } from "@wso2is/react-components";
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+    Form,
+    Grid,
+    Icon,
+    Modal,
+} from "semantic-ui-react";
+import { addUser, addUserRole, getGroupsList, getUserStoreList } from "../../api";
+
+/**
+ * Proptypes for the application consents list component.
+ */
+interface AddUserProps {
+    isBasicModalOpen: boolean;
+    handleModalClose: any;
+    isRoleModalOpen: boolean;
+    handleRoleModalOpen: any;
+    handleRoleModalClose: any;
+    getUserList: () => void;
+    onAlertFired: (alert: AlertInterface) => void;
+}
+
+/**
+ * Add user page.
+ *
+ * @return {JSX.Element}
+ */
+export const AddUser: React.FunctionComponent<AddUserProps> = (props: AddUserProps): JSX.Element => {
+
+    const [ groupsList, setGroupsList ] = useState([]);
+    const [ userStoreOptions, setUserStoresList ] = useState([]);
+    const [ username, setUsername ] = useState("");
+    const [ password, setPassword ] = useState("");
+    const [ domain, setDomain ] = useState("");
+    const [ roleIds, setRoleIds ] = useState([]);
+    const [ userId, setUserId ] = useState("");
+    const [ passwordOption, setPasswordOption ] = useState("");
+    const [ email, setEmail ] = useState("");
+
+    const {
+        getUserList,
+        isBasicModalOpen,
+        handleModalClose,
+        isRoleModalOpen,
+        handleRoleModalOpen,
+        handleRoleModalClose,
+        onAlertFired
+    } = props;
+    const { t } = useTranslation();
+    const passwordOptions = [
+        { label: "Ask password", value: "askPw" },
+        { label: "Create password", value: "createPw" },
+    ];
+
+    /**
+     * Fetch the list of available groups.
+     */
+    useEffect(() => {
+        getGroups();
+    }, []);
+
+    /**
+     * Fetch the list of available user stores.
+     */
+    useEffect(() => {
```

Comment:
Can't we use the above useEffect since both have same dependencies?

Reply:
The useEffect used to fetch the group list will be changed later.